### PR TITLE
text: adds memory and memory export

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This allows the WASI test to not require a checked in binary

### CODE REVIEW

File: wasi/testdata/args.wat [hunk 1]
```diff
@@ -9,7 +9,7 @@
 (module
   (import "wasi_snapshot_preview1" "args_sizes_get" (func $wasi_args_sizes_get (param i32 i32) (result i32)))
   (import "wasi_snapshot_preview1" "args_get"       (func $wasi_args_get (param i32 i32) (result i32)))
-  (memory 0x200)  ;; just an arbitrary size big enough for tests
```
Comment (codefromthecrypt) [inline wasi/testdata/args.wat:12]: punted hex for now

File: wasi/testdata/args.wat [hunk 2]
```diff
@@ -9,7 +9,7 @@
...
+  (memory 512)  ;; just an arbitrary size big enough for tests
```
Comment (mathetake) [inline wasi/testdata/args.wat:12]: this is the page size so maybe this is too big.. 😄  maybe single digits or at most tens is enough I think 😄

Comment (codefromthecrypt) [reply wasi/testdata/args.wat:12]: oh interesting. I was wondering and thought maybe it was bytes!

**Review by mathetake — APPROVED** [review]
This is so exciting!!!

**Review by nullpo-head — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
thanks for the looks @mathetake and @nullpo-head!

**Pull request merged**